src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -424,6 +420,28 @@ fn copy_self_contained_objects(
     target_deps
 }
 
+/// Resolves standard library crates for `Std::run_make` for any build kind (like check, build, clippy, etc.).
+pub fn std_crates_for_run_make(run: &RunConfig<'_>) -> Vec<String> {
+    // FIXME: Extend builder tests to cover the `crates` field of `Std` instances.
+    if cfg!(feature = "bootstrap-self-test") {
+        return vec![];
+    }
```

Comment:
Additional context:

https://github.com/rust-lang/rust/blob/72d73cec61aa8f85901358cd5d386d5dd066fe52/src/bootstrap/src/core/builder/tests.rs#L60-L62

That above and `doc_std` helper macros needs to be improved to mock `crates` field in `Std` with relevant values.